package.json

```diff
@@ -45,7 +45,8 @@
         "@astrojs/compiler": "^0.14.2",
         "debug": "^4.3.4",
         "eslint-visitor-keys": "^3.0.0",
-        "espree": "^9.0.0"
+        "espree": "^9.0.0",
+        "synckit": "^0.7.1"
```

Comment:
nice! We used the same package in our Prettier plugin because those APIs are synchronous. If more people need these to be synchronous, I think it would make sense to add these directly to our compiler.

Reply:
I found out about synckit by seeing the Prettier plugin using synckit. synckit works very well!

In my opinion, I don't think it is necessary to add a sync API to the compiler, for this parser and the Prettier plugin.
ESLint and Prettier don't yet support ESM plugins (though they are already planning). So if we add a sync API to the compiler, we still can't use it.
Also, the ESLint team is already planning an async parser. https://github.com/eslint/rfcs/pull/87
(However, I don't know about Prettier.)
I think it's best to use synckit until ESLint or Prettier support async plugins, and then stop using synckit once it supports async plugins.

If other ecosystems want a sync API, I think it's a good to add a sync API to the compiler.